### PR TITLE
LPFG-734 edit audience type

### PR DESCRIPTION
**Pull request by jakub-kainos**

### CODE REVIEW

File: src/controllers/audience/audienceController.ts [hunk 1]
```diff
@@ -149,30 +151,42 @@ export class AudienceController {
...
 			} else {
-				const savedAudience = await this.learningCatalogue.createAudience(req.params.courseId, audience)
-				req.session!.sessionFlash = {audience: savedAudience}
-				req.session!.save(() => {
-					res.redirect(
-						// prettier-ignore
-						`/content-management/courses/${req.params.courseId}/audiences/${savedAudience.id}/configure`
+				let savedAudienceId: string
+				if (res.locals.audience) {
+					this.audienceService.updateAudience(
```
Comment (elxsy) [inline src/controllers/audience/audienceController.ts]: MAINTAIN + READABILITY - Is there no concept of promises here rather callbacks?
`updateAudience` takes in a callback and then audience fed in again to call the same method. Hard to follow and see the execution in code let alone debugging.

The ideal scenario for readability.

```
audienceService.doSomething()
.then().doSomething()
.then().doOther()
.success(print)
.error(message)```

File: views/page/course/audience/audience-type.html [hunk 2]
```diff
@@ -6,30 +6,22 @@
 {% extends "../../../component/Page.njk" %}
 
 {% block pageTitle %}
-     {% if audience %}
-        {{ i18n.audience.page_title.edit_type }} - {{ i18n.proposition_name }}
-    {% else %}
-        {{ i18n.audience.page_title.add_type }} - {{ i18n.proposition_name }}
-    {% endif %}
+    {{i18n.audience.page_title.edit_type if audience else i18n.audience.page_title.add_type}} - {{i18n.proposition_name}}
```
Comment (Peter-D-Taylor) [inline views/page/course/audience/audience-type.html]: Page title isn't showing up in the add and configure audience pages

Comment (Peter-D-Taylor) [reply views/page/course/audience/audience-type.html]: Looks like it should be audience.pageTitle instead of page_title

File: src/controllers/audience/audienceController.ts (hunk 1 above)
Comment (jakub-kainos) [reply src/controllers/audience/audienceController.ts]: made changes removing the lambda - not using Futures, since it's not needed anymore due to the changes
